sdk/textanalytics/ai-text-analytics/test/public/utils/resultHelper.ts

```diff
@@ -3,14 +3,18 @@
 
 import { assert } from "chai";
 
-import { TextAnalyticsResult, TextAnalyticsSuccessResult } from "../../../src/";
+import { TextAnalyticsErrorResult, TextAnalyticsSuccessResult } from "../../../src/";
 
-export function assertAllSuccess(results: TextAnalyticsResult[]): void {
+export function assertAllSuccess<TSuccess extends TextAnalyticsSuccessResult>(
```

Comment:
Seems a little more complicated than it needs to be. Could just replace the TSuccess constraint with TextAnalyticsSuccessResult if you really want to remove TextAnalyticsResult, but see my other comment for why that's a breaking risk.

IIUC, `TSuccess extends TextAnalyticsSuccessResult` doesn't buy you much over `TextAnalyticsSuccessResult` unless you explicitly instantiate the generic whenever you call this or if you want to use extended object literals as arguments.

Reply:
I agree it does not buy much as is right now but it is precise and I always try to make my types as reasonably precise as possible so in the future we would like to make a change, we do not get constrained by the type and have to rethink it.

Reply:
Fair enough. I don't see the benefit of this vs. just using `TextAnalyticsSuccessResult` unless you explicitly instantiate the generic or use `as const` arguments everywhere it's used, but I have no problem with it.